assets/company-reports/actions.js

```diff
@@ -0,0 +1,82 @@
+import {errorHandler} from '../utils';
+import server from '../server';
+
+
+export const QUERY_REPORT = 'QUERY_REPORT';
+export function queryReport() {
+    return {type: QUERY_REPORT};
+}
+
+export const SET_ACTIVE_REPORT = 'SET_ACTIVE_REPORT';
+export function setActiveReport(data) {
+    return {type: SET_ACTIVE_REPORT, data};
+}
+
+export const RECEIVED_DATA = 'RECEIVED_DATA';
+export function receivedData(data) {
+    return {type: RECEIVED_DATA, data};
+}
+
+export const SET_ERROR = 'SET_ERROR';
+export function setError(errors) {
+    return {type: SET_ERROR, errors};
+}
+
+export function runReport() {
+    return function (dispatch, getState) {
+        dispatch(queryReport());
+        if (getState().activeReport == 'company-saved-searches') {
```

Comment:
this seems like mapping activeReport -> url so we could have a const dict:
```
const REPORTS = {'company-saved-searches': '/reports/componay/saved_searches', ...};
```
and later it would just get url and query:
```
const url = REPORTS[getState().activeReport];
dispatch(queryReport());
return server.get(url).then((data) => { dispatch(receivedData(data)));
```